### PR TITLE
feat(components): Add spacing variables

### PR DESCRIPTION
**Pull request by alisonjoseph**
https://github.ibm.com/Bluemix/carbon-issues/issues/410

Added initial spacing variables. Looking for feedback to make sure the way I set up the `_spacing.scss` file makes sense?

### Added
Added new variables to 
accordion
breadcrumb
button

###Questions
Does anyone see a need to have these variables included in the spacing scale? Or is it fine to just keep `center` and `auto`.
```
$spacing-center: center; 
$spacing-auto: auto;
```
Is this something that we will merge in as it gets done, or is it waiting for a v9 release?

### CODE REVIEW

**Review by joshblack — APPROVED** [review]
Looks awesome! Only comment would be if we want to add `spacing-none` or not. Totally understand if `0` is the preference, though 😄

File: src/globals/scss/styles.scss [hunk 1]
```diff
@@ -15,6 +15,7 @@ $css--plex: true !default;
 @import 'mixins';
 @import 'layout';
 @import 'layer';
+//@import 'spacing';
```
Comment (asudoh) [inline src/globals/scss/styles.scss]: Any plan to change here?

**Review by asudoh — COMMENTED** [review]
Good move, the only question I have is if we have plan to eliminate all hard-coded spacings. If not, we’d need our (internal) rule to use when to use the new variables and when not to. Thanks @alisonjoseph!

BTW I guess whether this change would bump up the major version depends on whether this change would affect any app using Carbon. (Do they have to make any code changes upon our change?)

File: src/globals/scss/styles.scss (hunk 1 above)
Comment (alisonjoseph) [reply src/globals/scss/styles.scss]: Oh yes, that shouldn't be commented out.

**Comment by alisonjoseph** [comment]
@asudoh Yes the plan is to standardize (update any outliers that don't fit on the new scale) and update the spacing on all of the components. I only added it to three components so far so it will be easier to review.

**Review by marijohannessen — APPROVED** [review]
Yay, this is so exciting! Works great! 👍

**Pull request merged**